src/common/utils.lua

```diff
@@ -241,48 +233,152 @@ function utils.camelCase(str)
 	return str
 end
 
-utils.notices = {}
+function utils.indexOf(t, value)
+	for i, v in ipairs(t) do
+		if v == value then
+			return i
+		end
+	end
+	return -1
+end
 
-function utils.notices.credit(msg)
-	local notice = {
-		Target = msg.From,
-		Action = "Credit-Notice",
-		Recipient = msg.Recipient,
-		Quantity = tostring(1),
+--- func description
+-- @param {table} params
+-- @param {string} params.id
+-- @param {number} params.timestamp
+-- @param {string | nil} params.owner
+-- @param {table | nil} params.controllers
+function utils.register(params)
+	assert(type(params) == "table", "Register Params must be a table")
+	local id, timestamp, owner, controllers = params.id, params.timestamp, params.owner, params.controllers
+	assert(type(id) == "string", "ANT ID must be a string")
+	assert(type(timestamp) == "number", "Timestamp must be a number")
+	assert(type(owner) == "string" or owner == nil, "Owner must be a string")
+	assert(type(controllers) == "table" or controllers == nil, "Controllers must be a table")
+	ANTS[id] = {
+		Owner = owner,
+		Controllers = controllers or {},
+		RegisteredAt = timestamp,
 	}
```

Comment:
```suggestion
ao.send({
   Target=id,
   Action="State"
   -- any other tags we want to forward
})
```

Reply:
actually, we just remove this util entirely. see other comments as to what the Handler should do, we don't need to do anything more than just forward a `State` request along.